golem_sci/implementation.py

```diff
@@ -588,19 +588,33 @@ def _pull_eth_subscription_events(self) -> None:
             subs = self._eth_subscriptions.copy()
         for sub in subs:
             try:
-                if sub.last_pulled_block >= self._confirmed_block:
-                    continue
-                transfers = self._find_incoming_eth_transfers(
-                    sub.last_pulled_block,
-                    self._confirmed_block,
-                    sub.address,
-                )
-                for t in transfers:
-                    self._on_event(t, sub.cb)
-                sub.last_pulled_block = self._confirmed_block
+                try:
+                    if sub.last_pulled_block >= self._confirmed_block:
+                        continue
+                    transfers = self._find_incoming_eth_transfers(
+                        sub.last_pulled_block,
+                        self._confirmed_block,
+                        sub.address,
+                    )
+                    for t in transfers:
+                        self._on_event(t, sub.cb)
+                    sub.last_pulled_block = self._confirmed_block
+                except ValueError as e:
+                    if (
+                            e.args
+                            and isinstance(e.args[0], dict)
+                            and 'missing trie node' in e.args[0].get('message')
+                    ):
+                        logger.warning(
+                            'Could not retrieve eth transfers: %r',
+                            e
+                        )
+                    else:
+                        raise
             except Exception as e:
                 logger.exception(
-                    'Exception while processing eth subscription: %r',
+                    'Unexpected exception '
```

Comment:
Aren't exceptions unexpected by definition?

Reply:
hmm... the `ValueError` above is kinda expected ... that would be the differentiation ;)